### DISCUSSION TITLE
Is UUIDv7 planned?

### DISCUSSION THREAD

**Discussion by VladislavRudakoff** [geldata/gel#7620] [Q&A]
I tried to find information on UUIDv7 both in the documentation and here on github. The only thing I found is these two issues, which mention that there are plans to switch from UUIDv1/4 to v7. ([5170](https://github.com/edgedb/edgedb/issues/5170#issuecomment-1856588489), [6538](https://github.com/edgedb/edgedb/issues/6538#issuecomment-1836906260))

I am interested in how soon these changes are planned to be implemented or whether they have been abandoned?

**Comment by CodesInChaos** [comment]
It was blocked by the UUIDv7 specification not being finalized yet. However it looks like [RFC 9562
](https://www.rfc-editor.org/rfc/rfc9562.html#name-uuid-version-7) was published in Mai 2024. So it should now be possible to implement it.

Implementation of v7 in postgres is work-in-progress: https://commitfest.postgresql.org/47/4388/ . It's tagged as postgres 17, which should release within the next three months or so. But since it apparently hasn't been merged yet, it's probably not guaranteed to make that release.

I expect edgedb to only add v7 support once it's in postgres.

From the July 23rd 2024 office hours:

> **Q:** If I currently use UUID v1mc as an ID, will migration to v7 be possible when it
>   is available?
>
> **Yury:**  The specification for UUID v7 is ready, so migration should be possible. While
> we haven’t finalized all the details of the migration process, it’s possible we
> will also have a detailed blog post about this transition.

**Reply by voxspox** [reply]
UUIDv7 is now available in [PostgreSQL 18](https://www.postgresql.org/docs/18/datatype-uuid.html)
